server/src/resolvers/event.js

```diff
@@ -1,10 +1,9 @@
 module.exports.Event = {
-  host: async (parent) => {
-  },
-  schedule: async (parent) => {
-  },
-  tags: async (parent) => {
-  },
-  participants: async (parent) => {
+  __resolveType(event, context, info) {
+    if (event.bookTitle) {
```

Comment:
This logic can be flipped to become cleaner code:

if (!event.bookTitle) {
   return null;
}

return 'EventBookClub';